src/test/xmeta/testlexer.cpp

```diff
@@ -1,76 +1,55 @@
-﻿#include "pch.h"
+#include "pch.h"
 #include <iostream>
 
 #include "antlr4-runtime.h"
 #include "XlangParser.h"
 #include "XlangLexer.h"
 #include "xlang_test_listener.h"
+#include "xmeta_idl_reader.h"
 #include <Windows.h>
 
 using namespace antlr4;
+using namespace xlang::xmeta;
 
 #pragma execution_character_set("utf-8")
 
-/* 
+/*
     Lexer test
-    Primarily test whether the antlr lexer is correctly tokenizing the string and the listener can 
-    store the correctly string. 
+    Primarily test whether the antlr lexer is correctly tokenizing the string and the listener can
+    store the correctly string.
 */
 
-
-/*
-    setup_and_run_parser
-
-    This helper method sets up the tokenizer and parser from the idl string and
-    walks through the AST with the xlang_test_listener. The test listener is used to checked
-    whether the idk string was lexed and parsed correctly by simply adding the string to a 
-    set which we check inside the check. This method also returns the number of syntax errors. 
-*/
-int setup_and_run_parser(std::string const& idl, xlang_test_listener &listener, bool disable_error_reporting = false)
+TEST_CASE("Namespace Identifier")
 {
-    ANTLRInputStream input(idl);
-    XlangLexer lexer(&input);
+    std::istringstream test_idl{ R"(
+        namespace test { }
+    )" };
 
-    CommonTokenStream tokens(&lexer);
-    XlangParser parser(&tokens);
-
-    if (disable_error_reporting) {
-        lexer.removeErrorListeners();
-        parser.removeErrorListeners();
-    }
-
-    tree::ParseTree *tree = parser.xlang();
-    tree::ParseTreeWalker::DEFAULT.walk(&listener, tree);
-    return parser.getNumberOfSyntaxErrors();
-}
-
-TEST_CASE("Namespace Identifier")
-{   
-    std::string test_idl =
-        "namespace test{}";
-        
     xlang_test_listener listener;
-    REQUIRE(setup_and_run_parser(test_idl, listener) == 0);
+    xmeta_idl_reader reader{ "" };
+    REQUIRE(reader.read(test_idl, listener) == 0);
 
-    std::set<std::string> &namespaces = listener.namespaces;
+    auto& namespaces = listener.namespaces;
```

Comment:
I don't really see a benefit of setting auto here. I guess its fine but yea.

Reply:
Just consistency. Shouldn't be in this PR though, we should have a separate commit for cleaning up the tests.

Reply:
It's no big deal just sneak it in haha